### PR TITLE
[WIP] Make VCC compile on Windows, feed with LLVM built with VS2022 using C…

### PR DESCRIPTION
**Pull request by devshgraphicsprogramming**
…langCL toolset (TODO: integrate it as part of https://github.com/Devsh-Graphics-Programming/Compiler-Explorer-DXC-Windows-Docker)

### CODE REVIEW

File: src/frontends/llvm/l2s_instr.c [hunk 1]
```diff
@@ -394,11 +394,11 @@ EmittedInstr convert_instruction(Parser* p, Node* fn_or_bb, BodyBuilder* b, LLVM
                     String name = get_string_literal(target->arena, name_node);
                     assert(name);
                     set_variable_name((Node*) target, name);
-                    return (EmittedInstr) {};
+                    //return {};
```
Comment (AnastaZIuk) [inline src/frontends/llvm/l2s_instr.c:397]: TODO: do not leave it without return, this thing requires attention and msvc doesn't like the previous statement

File: src/frontends/llvm/l2s_instr.c [hunk 2]
```diff
@@ -394,11 +394,11 @@ EmittedInstr convert_instruction(Parser* p, Node* fn_or_bb, BodyBuilder* b, LLVM
...
                 }
                 if (strcmp(intrinsic, "llvm.dbg.label") == 0) {
                     // TODO
-                    return (EmittedInstr) {};
+                    //return {};
```
Comment (AnastaZIuk) [inline src/frontends/llvm/l2s_instr.c:401]: TODO: do not leave it without return, this thing requires attention and msvc doesn't like the previous statement

**Review by AnastaZIuk — COMMENTED** [review]
TODO comments for me

**Comment by AnastaZIuk** [comment]
> …langCL toolset (TODO: integrate it as part of https://github.com/Devsh-Graphics-Programming/Compiler-Explorer-DXC-Windows-Docker)

also already integrated https://github.com/Devsh-Graphics-Programming/Compiler-Explorer-DXC-Windows-Docker/commit/433e6bf7c164a0e084cece9d9f2daffcf3b2b59e, we will make it running into production in 1 day

File: src/frontends/llvm/l2s_instr.c (hunk 1 above)
Comment (Hugobros3) [reply src/frontends/llvm/l2s_instr.c:397]: fixed in https://github.com/Hugobros3/shady/commit/14161c6f96cb3cfb86bc430d0e5a813b98d0b362

File: src/driver/vcc.c [hunk 3]
```diff
@@ -99,7 +99,11 @@ int main(int argc, char** argv) {
 
     info_print("built command: %s\n", arg_string);
 
+#ifdef WIN32
```
Comment (Hugobros3) [inline src/driver/vcc.c:102]: this type of stuff belongs in portability.c/h

File: src/frontends/llvm/CMakeLists.txt [hunk 4]
```diff
@@ -10,9 +10,15 @@ target_compile_definitions(shady_fe_llvm PRIVATE "LLVM_VERSION_MAJOR=${LLVM_VERS
 
 if (TARGET LLVM)
     message("LLVM shared library target exists, major version = ${LLVM_VERSION_MAJOR}")
-    target_link_libraries(shady_fe_llvm PRIVATE LLVM)
+	target_link_libraries(shady_fe_llvm PRIVATE LLVM)
```
Comment (Hugobros3) [inline src/frontends/llvm/CMakeLists.txt:13]: whitespace

File: src/frontends/llvm/CMakeLists.txt [hunk 5]
```diff
@@ -10,9 +10,15 @@ target_compile_definitions(shady_fe_llvm PRIVATE "LLVM_VERSION_MAJOR=${LLVM_VERS
...
 else ()
-    message(FATAL_ERROR "Failed to find LLVM target, but found LLVM module earlier")
+    llvm_map_components_to_libnames(LLVM support core)
```
Comment (Hugobros3) [inline src/frontends/llvm/CMakeLists.txt:15]: do you need support/core when linking against LLVM-C ? not sure...

File: src/shady/generator/CMakeLists.txt [hunk 6]
```diff
@@ -1,7 +1,17 @@
 find_package(json-c REQUIRED)
 
+if(WIN32)
```
Comment (Hugobros3) [inline src/shady/generator/CMakeLists.txt:3]: this is fine but if you want you can make it a global toggle instead to deal with all dependencies statically (where applicable)

**Comment by Hugobros3** [comment]
CI should work once you rebase

**Comment by Hugobros3** [comment]
pretty sure this is outdated now

**Pull request closed without merging**